python/cli/prelude_cli/views/shared.py

```diff
@@ -16,34 +14,30 @@ def handler(*args, **kwargs):
             click.secho(e.args[0], fg='red')
     return handler
 
-
-class Spinner:
+class Spinner(): 
     busy = False
-    delay = 0.1
 
-    @staticmethod
-    def spinning_cursor():
-        while 1:
-            for cursor in '|/-\\': yield cursor
-
-    def __init__(self, delay=None):
-        self.spinner_generator = self.spinning_cursor()
-        if delay and float(delay): self.delay = delay
+    def __init__(self, description='Loading'):
+       self.description = description
+       self.progress = Progress(
+            SpinnerColumn(style='green', spinner_name='line'),
+            TextColumn(f'[green]{description}'),
+            transient=True
+        )
 
     def spinner_task(self):
-        while self.busy:
-            sys.stdout.write(next(self.spinner_generator))
-            sys.stdout.flush()
-            time.sleep(self.delay)
-            sys.stdout.write('\b')
-            sys.stdout.flush()
-
+        with self.progress:
```

Comment:
True, I had that thought yesterday but wondered if it would lead to some weird behaviors where we're already calling Spinner. No reason to back that up though. Let me look into that